pkg/maintainer/relay.go

```diff
@@ -0,0 +1,85 @@
+package maintainer
+
+import (
+	"fmt"
+
+	"github.com/keep-network/keep-common/pkg/chain/ethereum"
+	"github.com/keep-network/keep-core/pkg/bitcoin"
+	"github.com/keep-network/keep-core/pkg/tbtc"
+	"github.com/mitchellh/mapstructure"
+	"github.com/spf13/viper"
+)
+
+type Config struct {
+	Bitcoin  bitcoin.Config
+	Ethereum ethereum.Config
+}
+
+// unmarshalConfig unmarshals config with viper from config file and command-line
+// flags into a struct.
+func unmarshalConfig(config *Config) error {
+	if err := viper.Unmarshal(
+		config,
+		viper.DecodeHook(
+			mapstructure.ComposeDecodeHookFunc(
+				mapstructure.StringToTimeDurationHookFunc(),
+				mapstructure.StringToSliceHookFunc(","),
+				mapstructure.TextUnmarshallerHookFunc(),
+			),
+		),
+	); err != nil {
+		return fmt.Errorf("failed to unmarshal configuration: %w", err)
+	}
+
+	return nil
+}
+
+// readConfigFile uses viper to read configuration from a config file. The config file
+// is not mandatory, if the path is
+func readConfigFile(configFilePath string) error {
+	// Read configuration from a file, located in `configFilePath`.
+	viper.SetConfigFile(configFilePath)
+
+	// Read configuration.
+	if err := viper.ReadInConfig(); err != nil {
+		return fmt.Errorf(
+			"failed to read configuration from file [%s]: %w",
+			configFilePath,
+			err,
+		)
+	}
+
+	return nil
+}
+
+func (c *Config) ReadConfig(configFilePath string) error {
+	// Read configuration from a file if the config file path is set.
+	if configFilePath != "" {
+		if err := readConfigFile(configFilePath); err != nil {
+			return fmt.Errorf(
+				"unable to load config (file: [%s]): [%w]",
+				configFilePath,
+				err,
+			)
+		}
+	}
+
+	// Unmarshal config based on loaded config file and command-line flags.
+	if err := unmarshalConfig(c); err != nil {
+		return fmt.Errorf("unable to unmarshal config: %w", err)
+	}
+
+	return nil
+}
+
+func NewRelay(btcChain bitcoin.Chain, tbtcChain tbtc.Chain) *Relay {
+	return &Relay{
+		btcChain:  btcChain,
+		tbtcChain: tbtcChain,
+	}
+}
+
+type Relay struct {
+	btcChain  bitcoin.Chain
+	tbtcChain tbtc.Chain
```

Comment:
I think this is wrong. Ideally, the `pkg/maintainer` should be independent of `pkg/tbtc`. That means we should declare our own `maintainer.RelayChain` interface with expectations regarding the chain required by the relay maintainer and implement it within `pkg/chain/ethereum`.

Reply:
Done in 1a02c7248bce4395be3830bc8b5bc4646f63f12a.